proxstar/__init__.py

```diff
@@ -142,14 +153,22 @@ def rq_dashboard_auth(*args, **kwargs):  # pylint: disable=unused-argument,unuse
 
 @app.errorhandler(404)
 def not_found(e):
-    user = User(session['userinfo']['preferred_username'])
-    return render_template('404.html', user=user, e=e), 404
+    try:
+        user = User(session['userinfo']['preferred_username'])
+        return render_template('404.html', user=user, e=e), 404
+    except KeyError as exception:
+        print(exception)
+        return render_template('404.html', user='chom', e=e), 404
```

Comment:
What happens when user='chom'? This is a user object that will be asked for attributes in base.html, so this may crash due to requesting an attribute from a string.

Reply:
Well, it _already_ crashes, that's why I tried to "fix" it. I guess we could set up an "unknown" user to return a 404 with.

Reply:
Can confirm that this does _not_ crash.